plugins/Synchro/SynchroSynth.cpp

```diff
@@ -0,0 +1,565 @@
+	connect(&m_modulatorDetune, SIGNAL(dataChanged()), this, SLOT(modulatorChanged()));
+	connect(&m_modulatorDrive, SIGNAL(dataChanged()), this, SLOT(modulatorChanged()));
+	connect(&m_modulatorSync, SIGNAL(dataChanged()), this, SLOT(modulatorChanged()));
+	connect(&m_modulatorChop, SIGNAL(dataChanged()), this, SLOT(modulatorChanged()));
+
+	//Update the general GUI when modulation controls are changed
+	connect(&m_modulation, SIGNAL(dataChanged()), this, SLOT(generalChanged()));
+	connect(&m_harmonics, SIGNAL(dataChanged()), this, SLOT(generalChanged()));
+	connect(&m_modulationStrength, SIGNAL(dataChanged()), this, SLOT(generalChanged()));
+}
+
+//Sends notes to be rendered and returns the result to LMMS
+void SynchroSynth::playNote(NotePlayHandle * n, sampleFrame * working_buffer)
+{
+	if (n->m_pluginData == NULL) { n->m_pluginData = new SynchroNote(n); }
+
+	const fpp_t frames = n->framesLeftForCurrentPeriod();
+	const f_cnt_t offset = n->noteOffset();
+
+	SynchroNote * ps = static_cast<SynchroNote *>(n->m_pluginData);
+
+	//Get pointers for sample-exact buffers
+	const ValueBuffer * modBuf = m_modulation.valueBuffer();
+	const ValueBuffer * modStrBuf = m_modulationStrength.valueBuffer();
+
+	for(fpp_t frame = offset; frame < frames + offset; ++frame)
+	{
+		sampleFrame outputSample = {0, 0};
+		sampleFrame tempSample = {0, 0};
+		SynchroOscillatorSettings carrier = { m_carrierDetune.value(), m_carrierDrive.value(),
+			m_carrierSync.value(), m_carrierChop.value(),
+			m_carrierAttack.value(), m_carrierDecay.value(), m_carrierSustain.value(), m_carrierRelease.value() };
+		SynchroOscillatorSettings modulator = { m_modulatorDetune.value(), m_modulatorDrive.value(),
+			m_modulatorSync.value(), m_modulatorChop.value(),
+			m_modulatorAttack.value(), m_modulatorDecay.value(), m_modulatorSustain.value(), m_modulatorRelease.value() };
+
+		//Oversampling using linear interpolation
+		for (int i = 0; i < SYNCHRO_OVERSAMPLE; ++i)
+		{
+			//Modulation strength and amount are sample-exact when Synchro can get the value from the buffer
+			float strength = modStrBuf ? modStrBuf->value(frame) : m_modulationStrength.value();
+			float amount = modBuf ? modBuf->value(frame) : m_modulation.value();
+
+			ps->nextStringSample(tempSample, Engine::mixer()->processingSampleRate() * SYNCHRO_OVERSAMPLE,
+				strength, amount, m_harmonics.value(), carrier, modulator);
+			outputSample[0] += tempSample[0];
+			outputSample[1] += tempSample[1];
+		}
+
+		for(ch_cnt_t chnl = 0; chnl < DEFAULT_CHANNELS; ++chnl)
+		{
+			working_buffer[frame][chnl] = outputSample[chnl] / SYNCHRO_OVERSAMPLE;
+		}
+	}
+	//applyRelease(working_buffer, n);
+	instrumentTrack()->processAudioBuffer(working_buffer, frames + offset, n);
+}
+
+//Tells LMMS how much extra time Synchro needs at the end of a note to finish our envelopes
+f_cnt_t SynchroSynth::desiredReleaseFrames() const
+{
+	return (f_cnt_t)(m_carrierRelease.value() * 0.001 * Engine::mixer()->processingSampleRate());
```

Comment:
Prefer C++-style casts over C-style casts. (`(f_cnt_t)` -> `static_cast<f_cnt_t>`)

Reply:
Fixed the cast you used as an example, making sure there are no other occurrences.